### DISCUSSION TITLE
Unable to recover 1wallet

### DISCUSSION THREAD

**Discussion by ttrenary** [polymorpher/one-wallet#231] [Q&A] [answered]
Hi - I apparently cleared all my browser data in ALL my browsers at some point and cannot restore my 1wallet with address: one1gtgd4e9atlk499arnwn4p3l99f2safhtlxtzxg 

It was created on 11/21 and I transferred a small amount of ONE from my main Harmony wallet one17dext3568agvs53vj2e4kmqetfwplhvydekwjg

Today I tried recovery using the first address and it would not recognize the QR code generated by the Google auth app. No errors - just totally didn't even think it was a QR code seemingly. I tried uploading the saved QR code (a .png file) and got a "Failed to parse QR code. Error: TypeError: The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type object" error.

Any help in recovering my wallet would be greatly appreciated.

**Comment by polymorpher** [comment, marked as answer]
Does the QR code have an address in the bottom? What does it look like, on high level?

**Reply by ttrenary** [reply]
It does, it is identical to the address I was trying to restore above. It looks just like a QR code with "Your 1wallet: <3 capitalized words here - not sure if I can share them publicly). Underneath this is the address. Thanks for helping, btw!

**Reply by polymorpher** [reply]
You are welcome. The QR code is not Google Auth QR code. It is the "address of wallet" QR code that lets other people send you money. It should look like the one below. If you are restoring on your computer, you can use the webcam to scan the QR code instead of uploading an image. See this video, for example: https://www.youtube.com/watch?v=zeOnRQmM-K8
![image](https://user-images.githubusercontent.com/913871/147521336-c4d2ad3c-7200-4426-ac76-2aa64ec12b8b.png)

**Reply by ttrenary** [reply]
Thank you! This worked to restore the wallet on a browser on my phone. However, I was unable to get it to work on my PC which seems to be due to a deficiency in my camera. Its inability to read the QR code for wallet export (under any lighting condition) is what caused be to go down the uploading route to begin with. I did try to grab screenshot of the code from my phone and transfer it it my PC but the process was too slow and the wallet would start to restore and then bomb out right when it was almost done - probably due to the timestamp info being out of sync. Looks like I'll need to install an authenticator app on my PC if I want to restore the wallet there, but for now I have access! Thank you again for your timely help! Go Harmony!

**Reply by polymorpher** [reply]
You are welcome. Glad to know it works. What was the spec of the computer? I want to make sure the process is not too slow for most computers.
